### DISCUSSION TITLE
8khz telephony audio ASR

### DISCUSSION THREAD

**Discussion by utkarshpandey12** [PaddlePaddle/PaddleSpeech#2144] [Q&A]
Hi,

Just wanted to know if Paddlespeech can detect speech from .wav audio files sampled at 8khz. If yes do I just staright away pass the audio to  using the CLI method or do I need to edit sample rate somewhere as pass it as a partameter? 

What is the correct approach to use PaddleSpeech for 8khz sampled .wav audio files?

Best
Utkarsh

**Comment by utkarshpandey12** [comment]
**Please suggest how to get started with transcribing 8khz .wav audio files using Python API or CLI option. 
Do I need to train the model from scratch as I see all the models have been trained using 16khz Audio data?.
As explained above I want to transcribe 8khz files either by using CLI/Python API.**

**When I am submitting 8khz file using the below CLI command it is returning me empty string however the audio file contains speech** 
( test) $paddlespeech asr --model transformer_librispeech --lang en --input test.wav

**I also tried giving the --sample_rate parameter = 8000 using the below command still it return empty string**
( test)  $paddlespeech asr --model transformer_librispeech --lang en --input test.wav --sample_rate 8000

@ryanrussell @harisankarh @ZeyuChen @wangkuiyi 

![Screenshot from 2022-07-12 21-30-33](https://user-images.githubusercontent.com/38834517/178539288-f38e1a7a-d45c-4442-9724-26bdffa65ba1.png)

**Comment by yt605155624** [comment]
we only support 16k ASR model till now, maybe you should train your own 8K ASR model or upsample your 8k wavs to 16k first and use our 16K models

**Reply by utkarshpandey12** [reply]
@yt605155624 

Can you briefly explain me the starting point to train model from scratch on my own 8khz dataset.
Which files to consider for training etc 
The documentation seems to be overwhelming for beginners to get started with training models from scratch.

Any help is appreciated 👍


Thanks

**Reply by yt605155624** [reply]
please check [callcenter](https://github.com/PaddlePaddle/PaddleSpeech/blob/develop/examples/callcenter/README.md)

**Reply by utkarshpandey12** [reply]
So is this flow of steps correct to finish with the training process? 
Please correct me if I am missing something.

**git clone 
cd examples 
cd callcenter
cd asr1
~/asr1
~/asr1 source ${MAIN_ROOT}/utils/parse_options.sh
~/asr1 bash run.sh --stage 0 --stop_stage 6**

**Reply by yt605155624** [reply]
Almost, but I suggest you run step by step to ensure that you know the function of each step
```bash
bash run.sh --stage 0 --stop_stage 0
bash run.sh --stage 1 --stop_stage 1
...
```

**Reply by utkarshpandey12** [reply]
**I started the 1st step in training the callcenter dataset using the command below** 
`bash run.sh --stage 0 --stop_stage 0`

**I'm getting  following errors. Can you please tell me what am I missing?**


center/asr1$ bash run.sh --stage 0 --stop_stage 0
checkpoint name conformer
mv: cannot stat 'data/manifest.train': No such file or directory
mv: cannot stat 'data/manifest.dev': No such file or directory
mv: cannot stat 'data/manifest.test': No such file or directory
----------- compute_mean_std.py Arguments -----------
delta_delta: 0
feat_dim: 80
manifest_path: data/manifest.train.raw
num_samples: -1
num_workers: 4
output_path: data/mean_std.json
sample_rate: 8000
spectrum_type: fbank
stride_ms: 10
target_dB: -20
use_dB_normalization: 0
window_ms: 25
-----------------------------------------------------------
2022-07-13 19:43:51.704 | INFO     | paddlespeech.s2t.frontend.augmentor.augmentation:__init__:122 - Augmentation: []
Traceback (most recent call last):
  File "/home/utkarsh/Desktop/peddle/PaddleSpeech/utils/compute_mean_std.py", line 87, in <module>
    main()
  File "/home/utkarsh/Desktop/peddle/PaddleSpeech/utils/compute_mean_std.py", line 77, in main
    normalizer = FeatureNormalizer(
  File "/home/utkarsh/Desktop/peddle/PaddleSpeech/paddlespeech/s2t/frontend/normalizer.py", line 117, in __init__
    self._compute_mean_std(manifest_path, featurize_func, num_samples,
  File "/home/utkarsh/Desktop/peddle/PaddleSpeech/paddlespeech/s2t/frontend/normalizer.py", line 166, in _compute_mean_std
    dataset = AudioDataset(manifest_path, num_samples, self._rng)
  File "/home/utkarsh/Desktop/peddle/PaddleSpeech/paddlespeech/s2t/frontend/normalizer.py", line 65, in __init__
    with jsonlines.open(manifest_path, 'r') as reader:
  File "/home/utkarsh/.pyenv/versions/peddle/lib/python3.9/site-packages/jsonlines/jsonlines.py", line 627, in open
    fp = builtins.open(file, mode=mode + "t", encoding=encoding)
FileNotFoundError: [Errno 2] No such file or directory: 'data/manifest.train.raw'
Compute mean and stddev failed. Terminated.

**Reply by zh794390558** [reply]
You need to prepare 8k train data yourself, and generate manifest for train,dev,test. You can see [this](https://github.com/PaddlePaddle/PaddleSpeech/blob/develop/dataset/aishell/aishell.py) as reference. Callcenter egs only has receipt to train, since we do not have 8k opensource data. Or you can using Aishell or Librispeech egs, change wav to 8k sample rate, then using model/preprocess config in Callcenter egs.

**Reply by utkarshpandey12** [reply]
Just want to confirm on one point 

Can I have training examples where there is no audio and for those audio files I will give an empty string in the dataset in the transcript txt file.
Or will it cause some issue ?
